handler/misc.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/prometheus/pushgateway/storage"
+)
+
+func HealthyHandler(
+	ms storage.MetricStore,
+) func(http.ResponseWriter, *http.Request, httprouter.Params) {
+	handlerFunc := prometheus.InstrumentHandlerFunc("healthy",
+		func(w http.ResponseWriter, _ *http.Request) {
+			healthy, err := ms.Healthy()
+			if healthy {
+				io.WriteString(w, "OK")
+			} else {
+				fmt.Fprintf(w, "FAIL: %v", err)
+			}
+		},
+	)
+	return func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
```

Comment:
Since you are not using the httprouter.Params, you can do this in an easier way, see the `Status` handler.